### DISCUSSION TITLE
Return params as an object in custom views instead of a segments array

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#14065] [Feature Requests & Ideas]
### Environment Info

Node.js v22.7.0

Binaries:
  Node: 22.7.0
  npm: 10.8.2
  Yarn: N/A
  pnpm: 9.12.1
Relevant Packages:
  payload: 3.0.0-beta.113
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.0.0: Tue Sep 24 23:39:07 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10

### Describe the Bug

In Payload v2, passing parameters to `path` in custom views (e.g. `/my-path/:id/test`) would pass the `params` as parsed by path-to-regexp to the child component. However, in v3 this was removed and even the example provided by `CustomViewWithParam` refer to the param by the index in the segments, instead of the named `id` parameter, which isn't ideal.

### Reproduction Steps

See https://github.com/payloadcms/payload/blob/86fdad0bb8ab27810599c8a32f3d8cba1341e1df/test/admin/components/views/CustomViewWithParam/index.tsx#L26 - in [the path config](https://github.com/payloadcms/payload/blob/a526c7becdf0309cf6069923c8d2ccd1f7a66a9f/test/admin/shared.ts#L11) there is an `:id` param, but the test does not / cannot make use of `id` being named, because only `segments` is returned.

**Comment by paulpopus** [comment]
Changed this into a discussion as it wasn't strictly an issue, we've had a change in functionality moving over from v2 but this is still a valid change request
